pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -47,6 +47,24 @@ func NewMetaProxier(ipv4Proxier, ipv6Proxier proxy.Provider) proxy.Provider {
 	})
 }
 
+// getProxierByIPFamily returns the proxy selected for a specific ipfamily
+func (proxier *metaProxier) getProxierByIPFamily(ipFamily v1.IPFamily) proxy.Provider {
+	if ipFamily == v1.IPv4Protocol {
+		return proxier.ipv4Proxier
+	}
+
+	return proxier.ipv6Proxier
+}
+
+func (proxier *metaProxier) getProxyByClusterIP(service *v1.Service) proxy.Provider {
+	ipFamily := v1.IPv4Protocol
+	if utilnet.IsIPv6String(service.Spec.ClusterIP) {
```

Comment:
so, `getProxyByClusterIP()` is always called for legacy services that only have one clusterIP, right?

Reply:
Yes. There is always a chance that (new-kube-proxy-code) talks to (old-api-server) and getting nil families. This happens during upgrade. This function works around that.

Reply:
Also stop calling them legacy service :-) they are current and will remain current for the foreseeable future.

Reply:
_until the feature gate is enabled by default_ :upside_down_face: